src/Bicep.LangServer/Handlers/BicepGenerateParamsCommandHandler.cs

```diff
@@ -33,41 +34,29 @@ namespace Bicep.LanguageServer.Handlers
     public class BicepGenerateParamsCommandHandler : ExecuteTypedResponseCommandHandlerBase<string, string>
     {
         private readonly ICompilationManager compilationManager;
-        private readonly IFeatureProviderFactory featureProviderFactory;
-        private readonly IFileResolver fileResolver;
-        private readonly IModuleDispatcher moduleDispatcher;
-        private readonly INamespaceProvider namespaceProvider;
-        private readonly IConfigurationManager configurationManager;
-        private readonly IApiVersionProviderFactory apiVersionProviderFactory;
-        private readonly IBicepAnalyzer bicepAnalyzer;
-
-        public BicepGenerateParamsCommandHandler(ICompilationManager compilationManager, ISerializer serializer, IFeatureProviderFactory featureProviderFactory, INamespaceProvider namespaceProvider, IFileResolver fileResolver, IModuleDispatcher moduleDispatcher, IConfigurationManager configurationManager, IApiVersionProviderFactory apiVersionProviderFactory, IBicepAnalyzer bicepAnalyzer)
+        private readonly BicepCompiler bicepCompiler;
+
+        public BicepGenerateParamsCommandHandler(ICompilationManager compilationManager, BicepCompiler bicepCompiler, ISerializer serializer)
             : base(LangServerConstants.GenerateParamsCommand, serializer)
         {
             this.compilationManager = compilationManager;
-            this.featureProviderFactory = featureProviderFactory;
-            this.namespaceProvider = namespaceProvider;
-            this.fileResolver = fileResolver;
-            this.moduleDispatcher = moduleDispatcher;
-            this.configurationManager = configurationManager;
-            this.apiVersionProviderFactory = apiVersionProviderFactory;
-            this.bicepAnalyzer = bicepAnalyzer;
+            this.bicepCompiler = bicepCompiler;
         }
 
-        public override Task<string> Handle(string bicepFilePath, CancellationToken cancellationToken)
+        public override async Task<string> Handle(string bicepFilePath, CancellationToken cancellationToken)
         {
             if (string.IsNullOrWhiteSpace(bicepFilePath))
             {
                 throw new ArgumentException("Invalid input file path");
             }
 
             DocumentUri documentUri = DocumentUri.FromFileSystemPath(bicepFilePath);
-            string output = GenerateCompiledParametersFileAndReturnOutputMessage(bicepFilePath, documentUri);
+            string output = await GenerateCompiledParametersFileAndReturnOutputMessage(bicepFilePath, documentUri);
 
-            return Task.FromResult(output);
+            return output;
```

Comment:
The compiler may be optimizing this away already, but replacing

```csharp
string output = await GenerateCompiledParametersFileAndReturnOutputMessage(bicepFilePath, documentUri);
return output;
```

with `return GenerateCompiledParametersFileAndReturnOutputMessage(bicepFilePath, documentUri);` would skip allocating and then resolving a local awaitable task.

Reply:
It's a perfectly valid point - I generally try to stay aware of whether I'm editing code which is performance-critical or not, and optimize for readability vs perf depending on where I'm at.

In this case, I don't see a great need to optimize (we've got much larger fish to fry!), so I didn't really give with/without `await` a second thought.